Otus-DevOps-2019-08/SergeyKa-cmd_infra
# Added files setupvpn.sh and cloud-bastion.ovpn

Added for further tests on Travis CI.

# Выполнено ДЗ № 5

 - [X] Main issue
 - [x] Task with *

## Roadmap:
 - Part 1
 + GCP account created. "bastionych" and "ellen-ripley" Vm instances are working correctly
 - Part 2
 + Prepared for two alternative shorthand path to connect with isolated "ellen-ripley" instance 
 - Part 3
 + Succesfull tests with connectivity to isolated "elen-ripley" instance using Pritunl VPN
 - Part 4
 + Prepared solution with trusted Certificate authority for Pritunl web interface

## How to launch project properly:
 - Prepare for ssh keys for Appuser user generating keys for GCP by using command:
 $ ssh-keygen -t rsa -f ~/.ssh/appuser -C appuser -P ""
 - Create instance with public IP with pre-installed VPN Pritunl web server (using attached setupvpn.sh file) and apply proper firewall inbound rules for this instance (Bastionych in that case) 
 - Create another instance without public IP connections which means to be isolated ("ellen-ripley" in that case) 
 - With generated file on Pritunl Web server (using  attached cloud-bastion.ovpn file) and OpenVPN client (used Pritunl Client in my case) connect to the isolated instance excluding to use instance with public IP addres (like host -> bastionych -> ellen-ripley) only using host -> ellen-ripley scheme.

## How to test:
 - Use attached cloud-bastion.ovpn file and Openvpn client on your host.
 - Try to connect with pre-installed ssh-key before and using command: 
 $ ssh -i ~/.ssh/appuser appuser@<host IP or hostname>
 - Ensure that your connection is established and secured on Pritunl Dashboard 35.187.97.192.nip.io

## PR checklist
 - [X] Выставил "cloud-bastion" с номером домашнего задания
 - [X] Выставил "cloud-bastion" с темой домашнего задания
